root/static/scripts/edit/utility/dates.js

```diff
@@ -11,9 +11,9 @@ import getDaysInMonth from '../../../../utility/getDaysInMonth.js';
 import parseInteger from '../../common/utility/parseInteger.js';
 
 type PartialDateWithStringsT = {
-  +day: StrOrNum | null,
-  +month: StrOrNum | null,
-  +year: StrOrNum | null,
+  +day?: StrOrNum | null,
+  +month?: StrOrNum | null,
+  +year?: StrOrNum | null,
```

Comment:
these should be `+day: ?StrOrNum,` etc. to allow the property to be null or undefined (rather than allowing the property to be omitted, or null if present)

Reply:
Shouldn't it be `+day?: ?StrOrNum` then? `dates.isDateValid({year: undefined, month: undefined, day: undefined})` and `dates.isDateValid({year: 2000})` are both expected to be `true` by the test.

Reply:
Yeah, `+day?: ?StrOrNum` is correct if both are valid.

Reply:
I'll type them as the test expects for now. Is there a way to automatically let Flow search for unneeded undefined / null options, for the time when we have everything typed?